### PR TITLE
[Fix]: Fix semantic segmentation related bug

### PR DESCRIPTION
**Pull request by Xiangxu-0103**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmdet3d/datasets/det3d_dataset.py [hunk 1]
```diff
@@ -229,15 +228,15 @@ def parse_data_info(self, info: dict) -> dict:
...
+        the absolute path. And process the `instances` field to
+        `ann_info` in training stage.
 
         Args:
             info (dict): Raw info dict.
 
         Returns:
             dict: Has `ann_info` in training stage. And
-            all path has been converted to absolute path.
+                all path has been converted to absolute path.
```
Comment (jshilong) [inline mmdet3d/datasets/det3d_dataset.py:239]: the original docstr format is correct

File: mmdet3d/datasets/kitti_dataset.py [hunk 2]
```diff
@@ -86,7 +86,7 @@ def parse_data_info(self, info: dict) -> dict:
 
         Returns:
             dict: Has `ann_info` in training stage. And
-            all path has been converted to absolute path.
+                all path has been converted to absolute path.
```
Comment (jshilong) [inline mmdet3d/datasets/kitti_dataset.py:89]: the original docstr format is correct
The format of `return` is different from the args

File: mmdet3d/datasets/scannet_dataset.py [hunk 3]
```diff
@@ -128,8 +128,8 @@ def parse_data_info(self, info: dict) -> dict:
             info (dict): Raw info dict.
 
         Returns:
-            dict: Data information that will be passed to the data
-            preprocessing transforms. It includes the following keys:
+            dict: Has `ann_info` in training stage. And
```
Comment (jshilong) [inline mmdet3d/datasets/scannet_dataset.py:131]: same here

File: mmdet3d/datasets/seg3d_dataset.py [hunk 4]
```diff
@@ -226,7 +224,7 @@ def parse_data_info(self, info: dict) -> dict:
 
         Returns:
             dict: Has `ann_info` in training stage. And
-            all path has been converted to absolute path.
+                all path has been converted to absolute path.
```
Comment (jshilong) [inline mmdet3d/datasets/seg3d_dataset.py:227]: same here

File: mmdet3d/datasets/s3dis_dataset.py [hunk 5]
```diff
@@ -166,24 +166,25 @@ class _S3DISSegDataset(Seg3DDataset):
...
+            as input. Defaults to dict(use_lidar=True, use_camera=False).
         ignore_index (int, optional): The label index to be ignored, e.g.
-            unannotated points. If None is given, set to len(self.CLASSES).
+            unannotated points. If None is given, set to len(self.CLASSES) to
+            be consistent with PointSegClassMapping function in pipeline.
             Defaults to None.
         scene_idxs (np.ndarray | str, optional): Precomputed index to load
             data. For scenes with many points, we may sample it several times.
             Defaults to None.
+        test_mode (bool, optional): Whether the dataset is in test mode.
```
Comment (ZCMax) [inline mmdet3d/datasets/s3dis_dataset.py:186]: we will not add `optional` for parameter whose default value is not None

File: mmdet3d/datasets/det3d_dataset.py [hunk 6]
```diff
@@ -28,42 +28,41 @@ class Det3DDataset(BaseDataset):
...
               outdoor point cloud 3d detection.
             - 'Depth': Box in depth coordinates, usually for
               indoor point cloud 3d detection.
             - 'Camera': Box in camera coordinates, usually
               for vision-based 3d detection.
-
-        filter_empty_gt (bool, optional): Whether to filter the data with
-            empty GT. Defaults to True.
-        test_mode (bool, optional): Whether the dataset is in test mode.
+        filter_empty_gt (bool): Whether to filter the data with empty GT.
```
Comment (ZCMax) [inline mmdet3d/datasets/det3d_dataset.py:54]: If it's set to be true, the example with empty annotations after data pipeline will be dropped and a random example will be chosen in `__getitem__`

File: mmdet3d/datasets/det3d_dataset.py [hunk 7]
```diff
@@ -28,42 +28,41 @@ class Det3DDataset(BaseDataset):
         ann_file (str): Annotation file path. Defaults to ''.
         metainfo (dict, optional): Meta information for dataset, such as class
             information. Defaults to None.
-        data_prefix (dict, optional): Prefix for training data. Defaults to
+        data_prefix (dict): Prefix for training data. Defaults to
             dict(pts='velodyne', img='').
         pipeline (list[dict], optional): Pipeline used for data processing.
```
Comment (ZCMax) [inline mmdet3d/datasets/det3d_dataset.py:33]: remove `optional`

File: mmdet3d/datasets/kitti_dataset.py [hunk 8]
```diff
@@ -20,11 +20,11 @@ class KittiDataset(Det3DDataset):
         ann_file (str): Path of annotation file.
         pipeline (list[dict], optional): Pipeline used for data processing.
             Defaults to None.
-        modality (dict, optional): Modality to specify the sensor data used
-            as input. Defaults to `dict(use_lidar=True)`.
+        modality (dict): Modality to specify the sensor data used as input.
+            Defaults to `dict(use_lidar=True)`.
         default_cam_key (str, optional): The default camera name adopted.
```
Comment (ZCMax) [inline mmdet3d/datasets/kitti_dataset.py:25]: remove `optional`

File: mmdet3d/datasets/transforms/dbsampler.py [hunk 9]
```diff
@@ -90,7 +90,7 @@ class DataBaseSampler(object):
         prepare (dict): Name of preparation functions and the input value.
         sample_groups (dict): Sampled classes and numbers.
         classes (list[str], optional): List of classes. Defaults to None.
-        points_loader(dict, optional): Config of points loader. Defaults to
+        points_loader (dict, optional): Config of points loader. Defaults to
```
Comment (ZCMax) [inline mmdet3d/datasets/transforms/dbsampler.py:93]: remove optional

File: mmdet3d/datasets/waymo_dataset.py [hunk 10]
```diff
@@ -23,8 +23,8 @@ class WaymoDataset(KittiDataset):
     Args:
         data_root (str): Path of dataset root.
         ann_file (str): Path of annotation file.
-        data_prefix (dict): data prefix for point cloud and
-            camera data dict. Default to dict(
+        data_prefix (dict, optional): data prefix for point cloud and
```
Comment (ZCMax) [inline mmdet3d/datasets/waymo_dataset.py:26]: remove optional

**Comment by ZCMax** [comment]
please fix the comments and lint problem, and LGTM

**Comment by Xiangxu-0103** [comment]
> @Xiangxu-0103 seems it still has some lint problem, please check it again

The lint problem has been fixed

**Pull request merged**